app/lib/actions/pulp3/orchestration/content_view_version/export.rb

```diff
@@ -0,0 +1,102 @@
+
+              action_output = plan_action(::Actions::Pulp3::ContentViewVersion::CreateExporter,
+                                          content_view_version_id: content_view_version.id,
+                                          smart_proxy_id: smart_proxy.id,
+                                          destination_server: destination_server).output
+
+              plan_action(::Actions::Pulp3::ContentViewVersion::Export,
+                          content_view_version_id: content_view_version.id,
+                          smart_proxy_id: smart_proxy.id,
+                          exporter_data: action_output[:exporter_data],
+                          chunk_size: chunk_size,
+                          from_content_view_version_id: from_content_view_version&.id)
+
+              plan_self(exporter_data: action_output[:exporter_data], smart_proxy_id: smart_proxy.id,
+                        destination_server: destination_server,
+                        content_view_version_id: content_view_version.id,
+                        from_content_view_version_id: from_content_view_version&.id)
+
+              plan_action(::Actions::Pulp3::ContentViewVersion::DestroyExporter,
+                          smart_proxy_id: smart_proxy.id,
+                          exporter_data: action_output[:exporter_data])
+            end
+          end
+
+          def run
+            smart_proxy = ::SmartProxy.find(input[:smart_proxy_id])
+            api = ::Katello::Pulp3::Api::Core.new(smart_proxy)
+            export_data = api.export_api.list(input[:exporter_data][:pulp_href]).results.first
+            output[:exported_file_checksum] = export_data.output_file_info
+            file_name = output[:exported_file_checksum].first&.first
+            path = File.dirname(file_name.to_s)
+            output[:export_path] = path
+            cvv = ::Katello::ContentViewVersion.find(input[:content_view_version_id])
+            from_cvv = ::Katello::ContentViewVersion.find(input[:from_content_view_version_id]) unless input[:from_content_view_version_id].blank?
+
+            export_metadata = ::Katello::Pulp3::ContentViewVersion::Export.new(
+                content_view_version: cvv,
+                smart_proxy: smart_proxy,
+                from_content_view_version: from_cvv).generate_metadata
+
+            toc_path_info = output[:exported_file_checksum].find { |item| item.first.end_with?("toc.json") }
+            export_metadata[:toc] = File.basename(toc_path_info.first)
+
+            history = ::Katello::ContentViewVersionExportHistory.create!(
+                content_view_version_id: input[:content_view_version_id],
+                destination_server: input[:destination_server],
+                path: path,
+                metadata: export_metadata
+            )
+            output[:export_history_id] = history.id
+            output[:path] = path
+          end
+
+          def humanized_name
```

Comment:
This should remain in the parent action?

Reply:
nope!